### PR TITLE
Replace action-rs/toolchain with manual config

### LINKED ISSUE #3833: Get rid of actions-rs/toolchain@v1
**Is your feature request related to a problem? Please describe.**
The github action `actions-rs/toolchain` hasn't been maintained for years now and is using features like `set-ouput` that are deprecated and might stop working anytime.

**Describe the solution you'd like**
Replace with something like:
```
steps:
  - name: Install rust
    run: |
      curl https://sh.rustup.rs -sSf | sh -s -- -y
      rustup default stable-${{ inputs.target }}
      rustup component add rustfmt
```

**Describe alternatives you've considered**
We could also use some alternative github actions like https://github.com/dtolnay/rust-toolchain

**Additional context**
Add any other context or information about the feature request here.

### PR DESCRIPTION
**Pull request by rdettai**
### Description

Closes #3833

### How was this PR tested?

- [x] Run ci.yaml
- [ ] Run publish_release_packages.yml

### CODE REVIEW

**Comment by fmassot** [comment]
@rdettai why not using https://github.com/dtolnay/rust-toolchain ?

**Comment by fmassot** [comment]
I'm okay with both. I don't have strong arguments on this.
@guilload, any preference?

**Comment by guilload** [comment]
`dtolnay/rust-toolchain` for conciseness.

**Comment by rdettai** [comment]
@guilload I took a deeper look into `dtolnay/rust-toolchain`. It's actually pretty confusing in many ways:
- it's quite unclear what the library's behavior is with `rust-toolchain.toml`:
  - Looks from the [issues](https://github.com/dtolnay/rust-toolchain/issues/29) that it is discarded
  - yet from what I see in the sources it's really just setting `rustup default xxx`
- the [way the action versions are mapped to toolchain versions](https://github.com/dtolnay/rust-toolchain#example-workflow) is very strange. They shouldn't be correlated, actions versions should be used to indicate breaking changes in the actions behavior itself.

If my understanding is right and the action doesn't support the override option, it means that:
```
    - name: Install rustup
      run: |
        curl https://sh.rustup.rs -sSf | sh -s -- -y
        rustup target add ${{ inputs.target }}
        rustup override set stable-${{ inputs.target }}
```

is written as: 
```
    - name: Install rustup
      uses: dtolnay/rust-toolchain@v1
      with:
        toolchain: stable
        targets: ${{ inputs.target }}
    - name: Set rustup override
      run: rustup override set ${{steps.toolchain.outputs.name}}
```
That's actually even longer as the explicit definition 😅 

With these observations, do you still believe we should use this action here?

**Comment by guilload** [comment]
rustup is fine too.

File: .github/actions/cargo-build-macos-binary/action.yml [hunk 1]
```diff
@@ -25,11 +25,12 @@ runs:
     - name: Install protoc
       run: brew install protobuf
       shell: bash
-    - uses: actions-rs/toolchain@v1
-      with:
-        toolchain: stable
-        targets: ${{ inputs.target }}
-        override: true
```
Comment (rdettai) [inline .github/actions/cargo-build-macos-binary/action.yml:32]: @guilload now that I think about it, why were we even overriding here in the first place? Doesn't it make more sense to build the releases with the toolchain versions pinned in `rust-toolchain.toml`?

Comment (guilload) [reply .github/actions/cargo-build-macos-binary/action.yml:32]: I don't know :) There was a time when we did not have a `rust-toolchain.toml` file. It should be the source truth indeed.

**Comment by rdettai** [comment]
@fmassot should we do the same tagging trick as for https://github.com/quickwit-oss/quickwit/pull/3832 to test the release workflows?

**Review by fmassot — APPROVED** [review]

**Pull request merged**